src/tools/wasm-as.cpp

```diff
@@ -116,6 +116,7 @@ int main(int argc, const char* argv[]) {
   }
 
   options.applyFeatures(wasm);
+  wasm.hasFeaturesSection = true;
```

Comment:
This seems fine to me. @kripken, wdyt?

Reply:
I think other tools might need this as well. I had similar problem in wasm-split today, although it's not in tests right now.

Reply:
Let's make the smallest change possible to get the tests working in this PR and discuss the broader change in a separate issue. Would you mind opening an issue about this?

Reply:
Sorry, I commented at the toplevel just now as I didn't realize this was the same topic as that.

I hope we can avoid adding this to a tool. Can we add `-all` to a test where we now need features enabled? If not then what specifically is the error we are running into?

Reply:
@kripken I tried to explain the situation in #3545.

Reply:
I think we'll probably want to add feature options to wasm-dis instead. Would you mind doing that as part of this PR or as a separate PR @martianboy?